packages/server/graphql/mutations/addMissingJiraField.ts

```diff
@@ -101,43 +108,51 @@ const addMissingJiraField = {
 
       return 0.5
     }
-    const possibleScreens = (
-      await Promise.all(
-        screens.map(async (screen) => {
-          const screenTabsResponse = await manager.getScreenTabs(cloudId, screen.id)
-          if (screenTabsResponse instanceof Error || screenTabsResponse instanceof RateLimitError) {
-            return null
-          }
-
-          const tabId = screenTabsResponse[0]?.id
-          return {screenId: screen.id, tabId, probability: evaluateProbability(screen)}
-        })
-      )
-    )
+
+    const possibleScreens = screens.map((screen) => {
+      return {screenId: screen.id, probability: evaluateProbability(screen), name: screen.name}
+    })
       .filter(isNotNull)
+      .filter((screen) => screen.probability >= 0.9)
       .sort((screen1, screen2) => screen2.probability - screen1.probability)
+
+    console.log(`Possible screens count: ${possibleScreens.length}`)
+    console.log('Possible screens:', JSON.stringify(possibleScreens))
+
     if (possibleScreens.length === 0) {
       return {error: {message: 'No screens available to modify!'}}
     }
 
     const dummyValues = {number: 0, string: '0'}
     const dummyValue = dummyValues[fieldType]
 
-    let updatedScreen: {screenId: string; tabId: string} | null = null
+    let updatedScreen: {screenId: string;} | null = null
     const screensToCleanup: Array<{screenId: string; tabId: string}> = []
     // iterate over all the screens sorted by probability, try to update the given field
     for (let i = 0; i < possibleScreens.length; i++) {
       const screen = possibleScreens[i]!
-      const {screenId, tabId} = screen
+      const {screenId} = screen
+
+      const screenTabsResponse = await manager.getScreenTabs(cloudId, screenId)
```

Comment:
Will add a comment and pagination. At the moment I saw a case with more than 700 screens.

We fetch only tabs in serial along with an attempt to add the field. 
However, we pick only screens with `probability >= 0.9` as other screens become to other projects.
This would be very few screens I guess. On a test project I saw only 2 matching screens.

Though, we can check later by logs to make sure.

Reply:
Will add a comment.

As we pick only screens with `probability >= 0.9` by default it will be only 2 screens. As others screens will be some custom screens, and we stop the iteration after a single screen is fixed — I expect the field should be fixed in a single iteration, so probably batching is not needed.

We can also analyse logs later to see real world numbers

Reply:
Added a comment